edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/PrometheusMetricsParser.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using Newtonsoft.Json;
+
+    public static class PrometheusMetricsParser
+    {
+        // Extracts data from prometheous format. Read about prometheous format here:
+        // https://prometheus.io/docs/concepts/data_model/
+        const string PrometheusMetricSchema = @"^(?<metricname>[^#\{\}]+)(\{((?<tagname>[^="",]+)=(\""(?<tagvalue>[^="",]+)\"")(,(?<tagname>[^="",]+)=(\""(?<tagvalue>[^="",]+)\""))*)\})?\s(?<metricvalue>.+)$";
+        static readonly Regex PrometheusSchemaRegex = new Regex(PrometheusMetricSchema, RegexOptions.Compiled);
+
+        public static IEnumerable<Metric> ParseMetrics(DateTime timeGeneratedUtc, string prometheusMessage)
+        {
```

Comment:
validate timeGeneratedUtc's DateKind is set to Utc; and return immediately if prometheusMessage is null or empty.

Reply:
It will return immediately if there are no lines to iterate over. I will add the utc check